sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -584,6 +602,11 @@
                 + ". If you are building with maven, set \"kafka.clients.version\" "
                 + "maven property to 0.10.1.0 or newer.");
       }
+      if (isCommitOffsetsInFinalizeEnabled()) {
+        checkArgument(getConsumerConfig().get(ConsumerConfig.GROUP_ID_CONFIG) != null,
```

Comment:
add a check to make sure `AUTO_COMMIT` is not set?

Reply:
I mentioned about it in javadoc for `commitOffsetsInFinalize()`, only either this option of AUTO_COMMIT is usually enabled, but not both. We don't need to explicitly need to enforce it. User could enable it (may be to debug, and sets very low frequency).
I will add a warning.